### DISCUSSION TITLE
Is there any chance technically to support old versions of android, and what theoretically can I do to make syncplay-mobile to be supported on Android 4.4.4?

### DISCUSSION THREAD

**Discussion by ivy-reps** [yuroyami/syncplay-mobile#90] [Q&A] [answered]
Also if it's possible, what exactly have to use developer who seeing kotlin for the first time? What IDE, what tools can be used for that?
I don't know whether I will be able to make it.
This idea or question came from I have old TV with Android 4.4.4. And Syncplay-mobile even of 0.4 -th versions can't be installed there because SDK is old. (doing corrections in Russian ChatGPT generated locale)

**Comment by yuroyami** [comment]
@ivy-reps For someone new to Android app development, it may be a little tricky to run the current code on your PC since it requires a few specific adjustments, but to answer your question: The tool/IDE needed is Android Studio (preferably the latest Beta version which is called 'Iguana' for the moment,it can be obtained from here [Android Studio Preview - Download](https://developer.android.com/studio/preview). You'll also need to have this [plugin](https://plugins.jetbrains.com/plugin/14936-kotlin-multiplatform-mobile) installed to the IDE because `syncplay-mobile` is currently written in `Kotlin Native multiplatform`.

I believe you shouldn't go through these steps as the code may not even be compatible with Android versions below Android 5 (Lollipop) but I am not sure. I will try to compile an APK that runs on Android 4.4 and we'll update you on this discussion.

**Reply by ivy-reps** [reply]
Thank you very much @yuroyami! If it's really simple I will be glad if you try. But please don't waste your time on that if it's not in  yours road map. Thank you very much anyway!

**Reply by ivy-reps** [reply]
> I believe you shouldn't go through these steps as the code may not even be compatible with Android versions below Android 5 (Lollipop) but I am not sure.

So, that means that probably if I will start to downgrade your code to make it to work on Android 4.4, AIWILLFAIL?
Yes I understand that you are using that Kotlin "adapter" which will allow you to make syncplay-mobile to work cross-platform which probably will make impossible to run it on Android 4.4

**Reply by yuroyami** [reply, marked as answer]
@ivy-reps I tried to make the app compatible with Android 4.4 but with no success unfortunately. The app has many dependencies on several libraries and it seems that many developers for those libraries ceased support for versions prior to Android 5.0 long ago. So it''s near impossible to make `syncplay-mobile` support anything below Android 5.0

**Comment by yuroyami** [comment]
With the recent releases, the minSdk has been bumped to Android 8.0, I know this is inconvenient for a number of users but maintaining old Android versions will be much more work than it is worth. My apologies
